### DISCUSSION TITLE
Saved Album Tracks filtered by Last.fm top artists

### DISCUSSION THREAD

**Discussion by BuckeyePay** [Chimildic/goofy#237] [Помощь с алгоритмом] [answered]
I am having a bit of logic problem. I am trying to create a playlist of tracks from my saved albums from a select number of my top artists from last.fm. What I don't know is how to get from artists to tracks, or if that is even possible.

let artists = Lastfm.getTopArtists({
  user: 'ChiliDog',
  period: '5year',
  limit: 25
});

not sure how to use this with getSavedAlbumTracks or getSavedAlbums based up the last.fm artists, or that is even something I can do.

**Comment by Chimildic** [comment, marked as answer]
1. Default `Lastfm.getTopArtists` don't give top for `5year` period. Because last.fm API have only such parameters: `overall`, `7day`, `1month`, `3month`, `6month`, `12month`. But goofy have [getCustomTop](https://chimildic.github.io/goofy/#/reference/lastfm?id=getcustomtop).
2. In order to keep albums only from top artists, we need to use [removeArtists](https://chimildic.github.io/goofy/#/reference/filter?id=removeartists) with `invert = true`.
3. In case with saved albums, tracks are located in response. Thus need to only `map` loop. With general albums probably require additional requests.

```js
function createPlaylistFromSavedAlbumsByTopArtists() {
  let savedAlbums = Source.getSavedAlbums()
  let topArtists = Lastfm.getCustomTop({
    type: 'artist',
    user: KeyValue.LASTFM_LOGIN,
    from: '2022-01-01',
    to: '2022-12-31',
    count: 25,
  })
  Filter.removeArtists(savedAlbums, topArtists, true)

  let albumsTracks = savedAlbums.map(a => a.tracks.items).flat()
  Playlist.saveWithReplace({
    name: 'Saved Albums From Top Artists',
    tracks: albumsTracks
  })
}
```

**Reply by BuckeyePay** [reply]
Thank you very much. You are so fast at responding to my stupid questions.
